### PR TITLE
[Button] Fix color icon red if button is plain and destructive

### PR DESCRIPTION
**Pull request by jacquesporveau**
### WHY are these changes introduced?
In cases that a button was passed the props destructive = true and plain
= true you would end up with a button that had text in one color (red /
destructive) and an icon in another color (blue / primary?).

This was not the intentional behaviour. I've added some CSS to color the
icon the same color as the destructive text.
Fixes https://github.com/Shopify/polaris-react/issues/2958

#### UI
| Before | After |
| ------- | ------ |
| Screen Shot 2020-05-09 at 9 50 12 PM | Screen Shot 2020-05-09 at 9 49 46 PM |

## 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {DeleteMinor} from '@shopify/polaris-icons';

import {Page} from '../src';
import {Button} from '../src/components/Button';

export function Playground() {
  return (
    <Page title="Playground">
      <Button destructive plain icon={DeleteMinor}>
        Delete
      </Button>
    </Page>
  );
}
```

</details>

### 🎩 checklist

* [ ] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
* [ ] Updated the component's `README.md` with documentation changes
* [ ] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
* [x] For visual design changes, pinged one of @ HYPD, @ mirualves, @ sarahill, or @ ry5n to update the Polaris UI kit

### CODE REVIEW

**Comment by jacquesporveau** [comment]
@dylanblanchard I've made some visual design changes.

File: UNRELEASED.md [hunk 1]
```diff
@@ -13,6 +13,7 @@
 
 - Added an outline on `Banner` for Windows high contrast mode ([#2878](https://github.com/Shopify/polaris-react/pull/2878))
 - Fixed Autocomplete / ComboBox focus ([#1089](https://github.com/Shopify/polaris-react/issues/1089))
+- Fixed coloring issue for button icons (#2958)
```
Comment (BPScott) [inline UNRELEASED.md:16]: Can this be a proper link please? Llike the line above

Comment (jacquesporveau) [reply UNRELEASED.md]: 👍

**Comment by jacquesporveau** [comment]
Thanks for the review @BPScott. Should be ready for another pass.

File: UNRELEASED.md [hunk 2]
```diff
@@ -14,6 +14,7 @@
 - Added an outline on `Banner` for Windows high contrast mode ([#2878](https://github.com/Shopify/polaris-react/pull/2878))
 - Fixed Autocomplete / ComboBox focus ([#1089](https://github.com/Shopify/polaris-react/issues/1089))
 - Fixed typing for EmptyState action ([#2977](https://github.com/Shopify/polaris-react/pull/2977))
+- Fixed coloring issue for button icons ([#2958](https://github.com/Shopify/polaris-react/issues/2958))
```
Comment (chloerice) [inline UNRELEASED.md:17]:
```suggestion
- Fixed incorrect `icon` color of `Button` when `destructive` and `plain`   ([#2958](https://github.com/Shopify/polaris-react/issues/2958))
```

File: UNRELEASED.md [hunk 3]
```diff
@@ -8,6 +8,8 @@
 
 ### Bug fixes
 
+- Fixed incorrect `icon` color of `Button` when `destructive` and `plain`   ([#2958](https://github.com/Shopify/polaris-react/issues/2958))
```
Comment (chloerice) [inline UNRELEASED.md:11]:
```suggestion
- Fixed incorrect `icon` color of `Button` when `destructive` and `plain` ([#2958](https://github.com/Shopify/polaris-react/issues/2958))
```

**Pull request merged**